packages/firestore/test/unit/lite-api/types.test.ts

```diff
@@ -695,4 +695,79 @@ describe('FirestoreTypeConverter', () => {
       expect(data.numberProperty).to.equal(43);
     }
   });
+
+  it('setDoc() fails to compile if `data` argument is missing properties', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<
+        { foo: string },
+        { bar: number }
+      >();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error `data` argument is missing `foo` property.
+      await setDoc(docRefWithConverter, { bar: 42 });
+    }
+  });
+
+  it('setDoc() fails to compile if `data` argument has incorrect type for a property', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<{ foo: string }, {}>();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error The `data` argument has the wrong type for `foo`.
+      await setDoc(docRefWithConverter, { foo: 42 });
+    }
+  });
+
+  it('updateDoc() fails to compile if `data` argument is missing properties', () => {
+    async function _(docRef: DocumentReference): Promise<void> {
+      const converter = new ThrowingConverter<
+        { foo: string },
+        { bar: number }
+      >();
+      const docRefWithConverter = docRef.withConverter(converter);
+      // @ts-expect-error `data` argument is missing `bar` property.
+      await updateDoc(docRefWithConverter, { foo: 'foo' });
```

Comment:
Nit: The error here should not be because `data` argument is missing `bar`. `bar` should be optional in `UpdateData<{bar: number}>`. Rather, the error should be that `foo` is not a member of `UpdateData<{bar: number}>`. This may affect the comment and test name.